test/Microsoft.NET.Build.Tests/GivenThatWeWantToVerifyNuGetReferenceCompat.cs

```diff
@@ -0,0 +1,227 @@
+        [InlineData("netstandard2.0", "Full", "netstandard1.0 netstandard1.1 netstandard1.2 netstandard1.3 netstandard1.4 netstandard1.5 netstandard1.6 netstandard2.0", true, true)]
+        [InlineData("netcoreapp1.0", "Full", "netstandard1.0 netstandard1.1 netstandard1.2 netstandard1.3 netstandard1.4 netstandard1.5 netstandard1.6 netcoreapp1.0", true, true)]
+        [InlineData("netcoreapp1.1", "Full", "netstandard1.0 netstandard1.1 netstandard1.2 netstandard1.3 netstandard1.4 netstandard1.5 netstandard1.6 netcoreapp1.0 netcoreapp1.1", true, true)]
+        [InlineData("netcoreapp2.0", "PartM1", "netstandard1.0 netstandard1.1 netstandard1.2 netstandard1.3 netstandard1.4 netstandard1.5 netstandard1.6 netcoreapp1.0 netcoreapp1.1 netcoreapp2.0", true, true)]
+        //  Fullframework NuGet versioning on Jenkins infrastructure issue
+        //        https://github.com/dotnet/sdk/issues/1041
+        //[InlineData("netcoreapp2.0", "Full", "netstandard1.0 netstandard1.1 netstandard1.2 netstandard1.3 netstandard1.4 netstandard1.5 netstandard1.6 netstandard2.0 netcoreapp1.0 netcoreapp1.1 netcoreapp2.0", true, true)]
+
+        //  OptIn matrix throws an exception for each permutation
+        //[InlineData("netstandard2.0", "OptIn", "net45 net451 net46 net461", true, true)]
+        //[InlineData("netcoreapp2.0", "OptIn", "net45 net451 net46 net461", true, true)]
+
+        public void Nuget_reference_compat(string referencerTarget, string testDescription, string rawDependencyTargets,
+                bool restoreSucceeds, bool buildSucceeds)
+        {
+            string referencerDirectoryIdentifierPostfix = "_" + referencerTarget + "_" + testDescription;
+
+            TestProject referencerProject = GetTestProject(ConstantStringValues.ReferencerBaseDirectory, referencerTarget, true);
+
+            foreach (string dependencyTarget in rawDependencyTargets.Split(',', ';', ' ').ToList())
+            {
+                TestProject dependencyProject = GetTestProject(ConstantStringValues.DependencyPrefix + dependencyTarget.Replace('.', '_'), dependencyTarget, true);
+                dependencyProject.PublishedNuGetPackageLibrary = new PackageReference(dependencyProject.Name, "1.0.0", ConstructNuGetPackageReferencePath(dependencyProject));
+                referencerProject.ReferencedProjects.Add(dependencyProject);
+            }
+
+            //  Skip running .NET Framework tests if not running on Windows
+            //        https://github.com/dotnet/sdk/issues/335
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (!AllProjectsBuildOnNonWindows(referencerProject))
+                {
+                    return;
+                }
+            }
+
+            //  Set the referencer project as an Exe unless it targets .NET Standard
+            if (!referencerProject.ShortTargetFrameworkIdentifiers.Contains(ConstantStringValues.NetstandardToken))
```

Comment:
'ShortTargetFrameworkIdentifiers' is a list of strings; so we need to use the 'contains'. In the current usage, the list contains only one. However, the test is able to handle a multi-targeted reference project.